### PR TITLE
All: add allowed_id[] parameter for journeys

### PR DESCRIPTION
**Pull request by TeXitoi**

### CODE REVIEW

File: source/jormungandr/jormungandr/scenarios/default.py [hunk 1]
```diff
@@ -130,9 +130,10 @@ def parse_journey_request(self, requested_type, request):
             req.journeys.streetnetwork_params.origin_mode = "bss"
         if req.journeys.streetnetwork_params.destination_mode == "bike_rental":
             req.journeys.streetnetwork_params.destination_mode = "bss"
-        if "forbidden_uris[]" in request and request["forbidden_uris[]"]:
-            for forbidden_uri in request["forbidden_uris[]"]:
-                req.journeys.forbidden_uris.append(forbidden_uri)
+        for forbidden_uri in get_or_default(request, "forbidden_uris[]", []):
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/scenarios/default.py:133]: I'm not sure what `get_or_default` do, but I think you can just do 
`for forbidden_uri in request.get("forbidden_uris[]", []):`

File: source/routing/raptor.cpp [hunk 2]
```diff
@@ -520,88 +526,115 @@ RAPTOR::isochrone(const map_stop_point_duration& departures,
...
+    boost::dynamic_bitset<> valid_journey_pattern_points(jp_container.nb_jpps());
+    valid_journey_pattern_points.set();
+    valid_stop_points.set();
+
+    auto forbidden_objs = ObjsFromIds(forbidden, jp_container, data);
+    valid_journey_patterns &= forbidden_objs.jps.flip();
+    valid_journey_pattern_points &= forbidden_objs.jpps.flip();
+    valid_stop_points &= forbidden_objs.sps.flip();
+
+    const auto allowed_objs = ObjsFromIds(allowed, jp_container, data);
```
Comment (antoine-de) [inline source/routing/raptor.cpp:630]: you can maybe add a comment here explaining the mechanism

File: source/jormungandr/jormungandr/scenarios/default.py (hunk 1 above)
Comment (TeXitoi) [reply source/jormungandr/jormungandr/scenarios/default.py:133]:
```python
def get_or_default(request, val, default):
    """
    Flask add all API param in the request dict with None by default,
    so here is a simple helper to get the default value is the key
    is not in the dict or if the value is None
    """
    val = request.get(val, default)
    if val is not None:
        return val
    return default
```

Looks like I can't

File: documentation/slate/source/includes/apis.md [hunk 3]
```diff
@@ -935,6 +936,43 @@ The [isochrones](#isochrones) service exposes another response structure, which
 | nop     | wheelchair           | boolean | If true the traveler is considered to be using a wheelchair, thus only accessible public transport are used<br>be warned: many data are currently too faint to provide acceptable answers with this parameter on       | False       |
 | nop     | debug                | boolean | Debug mode<br>No journeys are filtered in this mode     | False       |
 
+### Precisions on `forbidden_uris[]` and `allowed_id[]`
+
+These parameters are filtering the vehicle journeys and the stop points use to compute the journeys.  The journeys can only use allowed vehicle journeys (as present in the `public_transport` or `on_demand_transport` sections). They also can only use the allowed stop points (as present in the `street_network`, `waiting` and `crow_fly` sections).
```
Comment (antoine-de) [inline documentation/slate/source/includes/apis.md:941]: :+1: seems quite clear for me.

Hovewer I don't find the hand drawn picture cristal clear, I think a computer generated schema would be better

File: source/routing/raptor.cpp (hunk 2 above)
Comment (antoine-de) [reply source/routing/raptor.cpp:630]: done

File: documentation/slate/source/includes/apis.md (hunk 3 above)
Comment (TeXitoi) [reply documentation/slate/source/includes/apis.md:941]: if you don't like the new one, tune it yourself:
```dot
graph G {
  rankdir=LR;
  SPA -- SPB [label="LA"];
  SPA -- SPB [label="LB"];
  SPA -- SPC [label="LC"];
  SPA -- SPC [label="LD"];
  SPC -- SPB [label="LE"];
  SPC -- SPB [label="LF"];
}
```

**Review by antoine-de — APPROVED** [review]

**Pull request merged**